OpenSim/Simulation/Model/Geometry.h

```diff
@@ -0,0 +1,641 @@
+ * Author(s): Ayman Habib                                                     *
+ *                                                                            *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may    *
+ * not use this file except in compliance with the License. You may obtain a  *
+ * copy of the License at http://www.apache.org/licenses/LICENSE-2.0.         *
+ *                                                                            *
+ * Unless required by applicable law or agreed to in writing, software        *
+ * distributed under the License is distributed on an "AS IS" BASIS,          *
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.   *
+ * See the License for the specific language governing permissions and        *
+ * limitations under the License.                                             *
+ * -------------------------------------------------------------------------- */
+
+#include "SimTKcommon.h"
+#include <OpenSim/Common/Component.h>
+#include <OpenSim/Common/Set.h>
+#include "ModelComponent.h"
+#include "Appearance.h"
+
+namespace OpenSim { 
+
+class PhysicalFrame;
+class Model;
+class Frame;
+class ModelComponent;
+//=============================================================================
+//=============================================================================
+/**
+ Class Geometry is intended to be used as the base class for all
+ geometry that needs to be represented in the system, including mesh files, 
+ and built in analytic shapes. Any ModelComponent can specify a list of
+ Geometry items to represent itself in graphics window. The relation between
+ a ModelComponent and specific Geometry utilizes the Component mechanism, as
+ the specific pieces of geometry are treated as subcomponents. The placement
+ of the Geometry in 3D spae is computed from the Frame that the Geometry is
+ "Connected" to.
+
+ Geometry (and all its subclasses) serve as the set of higher level primitives
+ available to OpenSim component writers to express the Geometry of interest.
+ The Geometry class handles serialization and also the translation to a set of
+ DecorativeGeometry objects that gets passed to the Visualization system to be
+ rendered. 
+
+ @author Ayman Habib
+ */
+
+// Base Geometry
+class OSIMSIMULATION_API Geometry : public Component {
+    OpenSim_DECLARE_ABSTRACT_OBJECT(Geometry, Component);
+public:
+    // Properties common to all Geometry types are included as Properties
+    // of the base class.
+    // Scale factors
+    OpenSim_DECLARE_PROPERTY(scale_factors, SimTK::Vec3,
+        "Scale factors in X, Y, Z directions respectively.");
+    // Default display properiies e.g. Representation, color, texture, etc.
+    OpenSim_DECLARE_UNNAMED_PROPERTY(Appearance,
+        "Default appearance for this Geometry");
+
+    enum DisplayPreference {
+        Hide = 0,       ///<Hide geometry from display
```

Comment:
I am confused. Why is the DisplayPreference not a part of Appearance? For me the Draw prefix adds little. `Hidden`, `Points`, `WireFrame`, `Surface` seem clear enough. Why is an explicit `Default` required is that not implied if one is not specified?